lib/logstash/inputs/snmptrap.rb

```diff
@@ -47,26 +47,44 @@ def register
     end
   end # def register
 
-  public
   def run(output_queue)
     begin
       # snmp trap server
       snmptrap_listener(output_queue)
     rescue => e
       @logger.warn("SNMP Trap listener died", :exception => e, :backtrace => e.backtrace)
-      sleep(5)
-      retry
+      stoppable_call(20) { sleep(0.25) }
```

Comment:
yep

```ruby
jruby-9.0.0.0 :001 > a = Thread.new { 1} 
 => #<Thread:0x77ec78b9 dead> 
jruby-9.0.0.0 :002 > a.wakeup
ThreadError: killed thread
```

But as @colinsurprenant said, let's discuss this in an issue for this purpose, there's already "potentially lost" valuable content in here.